controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -1425,6 +1440,92 @@ bool master_thread::handle_intel_slave_join(Socket *sd, ieee1905_1::CmduMessageR
+                                                ieee1905_1::CmduMessageTx &cmdu_tx)
+{
+    // TODO backhaul handling.
+    // Multi-AP Agent doesn't say anything about the backhaul, so we have to rely on Intel Slave Join for that.
+    std::string backhaul_mac = "";
+
+    // TODO bridge handling.
+    // Assume repeater
+    beerocks::eType ire_type = beerocks::TYPE_IRE;
+
+    // bridge_mac node may have been created from DHCP/ARP event, if so delete it
+    // this may only occur once
+    if (database.has_node(bridge_mac) && (database.get_node_type(bridge_mac) != ire_type)) {
+        database.remove_node(bridge_mac);
+    }
+    // add new GW/IRE bridge_mac
+    LOG(DEBUG) << "adding node " << bridge_mac << " under " << backhaul_mac << ", and mark as type "
+               << ire_type;
+    database.add_node(bridge_mac, backhaul_mac, ire_type);
+    database.set_node_state(bridge_mac, beerocks::STATE_CONNECTED);
+
+    // Controller expects backhaul which contains bridge which contains radio, but we only get radio...
+
+    // Update existing node, or add a new one
+    if (database.has_node(radio_mac)) {
+        if (database.get_node_type(radio_mac) != beerocks::TYPE_SLAVE) {
+            database.set_node_type(radio_mac, beerocks::TYPE_SLAVE);
+            LOG(ERROR) << "Existing mac node is not TYPE_SLAVE";
+        }
+        database.clear_hostap_stats_info(radio_mac);
+    } else {
+        // TODO Intel Slave Join has separate radio MAC and UID; we use radio_mac for both.
+        database.add_node(radio_mac, bridge_mac, beerocks::TYPE_SLAVE, radio_mac);
+    }
+    database.set_hostap_is_acs_enabled(radio_mac, false);
+    // TODO
+    //        if (!notification->is_slave_reconf()) {
+    //            son_actions::set_hostap_active(database, tasks, radio_mac,
+    //                                           false); // make sure AP is marked as not active
+    //        }
+
+    // TODO Assume repeater mode
+    database.set_hostap_repeater_mode_flag(radio_mac, true);
+    // TODO Assume no backhaul manager
+    database.set_hostap_backhaul_manager(radio_mac, false);
+
+    database.set_node_state(radio_mac, beerocks::STATE_CONNECTED);
+    database.set_node_backhaul_iface_type(radio_mac, beerocks::IFACE_TYPE_BRIDGE);
+    // TODO iface_name will not be set
+    // TODO driver_version will not be set
+    database.set_hostap_iface_type(radio_mac, IFACE_TYPE_WIFI_UNSPECIFIED);
+
+    // TODO number of antennas comes from HT/VHT capabilities (implicit from NxM)
+    // TODO ant_gain and conducted_power will not be set
+    database.set_node_name(radio_mac, tlvwscM1->model_name());
+    // TODO ipv4 will not be set
+
+    // sd is assigned to src bridge mac
+    sd->setPeerMac(bridge_mac);
+
+    // TODO database.set_hostap_supported_channels from radio_basic_caps->operating_classes_info_list
+    // TODO assume SSIDs are not hidden
+    database.set_hostap_advertise_ssid_flag(radio_mac, true);
+
+    // TODO
```

Comment:
WSC autoconfig message contains the capabilities in the WSC M1 RF_Bands attribute:
![image](https://user-images.githubusercontent.com/13386849/63024133-46546100-beaf-11e9-883b-31e9dd100657.png)
We should use this information here.

Reply:
> WSC autoconfig message contains the capabilities in the WSC M1 RF_Bands attribute:
> We should use this information here.

I skipped that because in the current database, this information is not kept separately. Instead, it is constructed out of the channel information. We should get the channel information from the AP Capability Report message.